### ISSUE TITLE
build-wasm throws error for a new tree-sitter grammar

### ISSUE THREAD

**Issue by ganezdragon** [tree-sitter/tree-sitter#879]
hi, I'm writing a grammar for perl in tree-sitter and have used this doc - https://github.com/tree-sitter/tree-sitter/blob/master/lib/binding_web/README.md, to generate .wasm file.

But I'm getting the following error
```
sudo npx tree-sitter build-wasm node_modules/tree-sitter-perl/

emcc command failed - docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"mkdir /src/tree-sitter-perl: file exists\"": unknown.
```

My perl grammar - https://github.com/ganezdragon/tree-sitter-perl

I have tried running it for a `tree-sitter-javascript` grammar, and it works fine, and the .wasm file is generated for that.

Can anyone help me out here?

**Comment by ahlinc** [comment]
@maxbrunsfeld it seems a bug that only happens when a docker version of `emcc` is used and happen on a docker container creation stage. May be a workaround like mounting a real system path with all resolved symlinks may help.
The issue happens when a local repo folder was used to install as the `npm` package.
Steps to reproduce:
```bash
mkdir foobar
cd foobar/
git clone https://github.com/ganezdragon/tree-sitter-perl
npm init -y
npm i -D tree-sitter-cli ./tree-sitter-perl/
npx tree-sitter build-wasm node_modules/tree-sitter-perl/
```

@ganezdragon For now as a workaround you may use a one line command like:
```bash
cd node_modules/tree-sitter-perl/ && npx tree-sitter build-wasm && cd - && mv node_modules/tree-sitter-perl/*.wasm .
```

**Comment by ganezdragon** [comment]
@maxbrunsfeld, sorry I forgot about this thread. Thanks for response. I'm yet to try your method, maybe someone could use if they come across this issue.

Eventually I tried this method and it worked,

- I ran this sh script https://github.com/emscripten-core/emsdk/blob/0eb40e10a25570ba34a9ff8116c39c86a97474a2/emsdk_env.sh#L0-L1
- And then ran the following, `tree-sitter build-wasm node_modules/tree-sitter-perl/`

This did the trick for me.
